packages/next-app/src/app/stableswapsimulator/(components)/BaselineDataForm.tsx

```diff
@@ -0,0 +1,99 @@
+"use client";
+
+import { StableSwapSimulatorDataSchema } from "@balancer-pool-metadata/schema";
+import { capitalize } from "@balancer-pool-metadata/shared";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+
+import { Input } from "#/components/Input";
+import { AnalysisData, useStableSwap } from "#/contexts/StableSwapContext";
+import useDebounce from "#/hooks/useDebounce";
+
+import { TokenTable } from "./TokenTable";
+
+export default function BaselineDataForm() {
+  const { baselineData, setBaselineData, newPoolImportedFlag } =
+    useStableSwap();
+  const {
+    register,
+    getValues,
+    clearErrors,
+    watch,
+    formState: { errors },
+  } = useForm<typeof StableSwapSimulatorDataSchema._type>({
+    resolver: zodResolver(StableSwapSimulatorDataSchema),
+    mode: "onChange",
+  });
+
+  const swapFee = watch("swapFee");
+  const ampFactor = watch("ampFactor");
+  const tokens = watch("tokens");
+  const debouncedSwapFee = useDebounce(swapFee);
+  const debouncedAmpFactor = useDebounce(ampFactor);
```

Comment:
It almost looks this defeats the purpose of having a form in the first place. Same with the setting of params we do in the context through `handleChange`. A much more elegant solution IMO would be triggering the form submission remotely -- e.g. define `const submit = useDebounce(handleSubmit(onSubmit))` and call `submit()` when you want to submit it. It leverages more of the form logic instead of reimplementing it

Reply:
I also thought to remove the form (since we are dealing with changes as soon as they happen) but I want to reuse our validations. Will look for the solution that we described.